File: js/flattenArray.js
function getFlattenArray() {
  var newArray = [];
	function flattenArray(arr, res) {
  for(let idx = 0; idx < arr.length; idx++) {
    if(Array.isArray(arr[idx])) {
      flattenArray(arr[idx], res)
    } else {
      res.push(arr[idx])
    }
  }
} 
  flattenArray(arr, newArray)
  return newArray;
}

File: recursion/flattenArray.py
def flattenArray(arr, res):
  for idx in range(len(arr)):
    if isinstance(arr[idx], list):
      flattenArray(arr[idx], res)
    else:
      res.append(arr[idx])
  return res

print(flattenArray([[1], [2, 3], [4], [3, [2, 4]]], []))


